### ISSUE TITLE
[Notification] Drop the maintenance of Node.js 0.10 and 0.12 for Azure Storage  Client Library for Node.js

### ISSUE THREAD

**Issue by hasonmsft** [Azure/azure-storage-node#249]
Hi All,

Starting from Jan, 2017, Node.js 0.10 and 0.12 will be out of the Node.js’ [official LTS schedule](https://github.com/nodejs/LTS#lts-schedule). To align with Node.js plan, we will no longer maintain Azure Storage Client Library for Node.js for Node.js 0.10 and 0.12 versions. You may still be able to use our latest SDK with Node.js 0.10 and 0.12 however note that we no longer validate it against Node.js 0.10 and 0.12 and no bug fix will be done for these 2 versions from our side. 

Haibo Song,
Azure Storage Team

**Comment by peterblazejewicz** [comment]
This is already enforced in Travis CI configuration:
https://github.com/Azure/azure-storage-node/blob/master/.travis.yml#L3-L6
Can I go with NPM package update:
https://github.com/Azure/azure-storage-node/blob/master/package.json#L19
How this could be verified? I've already run tests with 6.* on OS X.

**Comment by hasonmsft** [comment]
@peterblazejewicz , what's your test result with 6.* on OS X? As the package.json specified, 0.8.26+ node are still valid to use our package including 6.*. But we won't verify 0.8.*. 0.10.*, 0.12.*

Thanks,
Haibo

**Comment by peterblazejewicz** [comment]
When running localy with mocks, I always get:
```bash
  516 passing (19s)
  107 pending
```
Details are here:
https://gist.github.com/peterblazejewicz/d54abb625a1f54148d1b4360d6dcf471

**Comment by hasonmsft** [comment]
It looks good. The pending cases are because they can only be run against live storage account. Under mock mode, it'll be skipped (pending).

**Comment by peterblazejewicz** [comment]
Can the `package.json` be updated then?
```diff
-    "node": ">= 0.8.26"
+    "node": ">=4"
```
That's the starting point to move things, as the Node < v4 is no longer maintained starting 2016-12-31.
Thanks!

**Comment by jeffwilcox** [comment]
Also, the contribution guidelines for this repo still note the requirement to be v0.10 compatible, so those can be updated...

https://github.com/Azure/azure-storage-node/blob/master/CONTRIBUTING.md#adding-features-for-all-platforms

**Comment by XiaoningLiu** [comment]
Thanks! ```package.json``` can be updated in next **major** version release. Because it's kind of breaking change. The contribution guidelines will be updated.

**Comment by XiaoningLiu** [comment]
Fix has been made in 2.8.2 release, thanks!
